### PR TITLE
syslog: T5367: add format option to include timezone in message

### PR DESCRIPTION
**Pull request by c-po**
## Change Summary

Add CLI option to include the systems timezone in the syslog message sent to a collector. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5367

## Component(s) name

syslog

## How to test

`set system syslog host <hostname> format include-timezone`

## Smoketest result

```
cpo@LR1.wue3:~$ /usr/libexec/vyos/tests/smoke/cli/test_system_syslog.py
test_syslog_basic (__main__.TestRSYSLOGService.test_syslog_basic) ... ok

----------------------------------------------------------------------
Ran 1 test in 3.977s

OK
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by dmbaturin — APPROVED** [review]

**Review by jestabro — APPROVED** [review]

File: data/templates/rsyslog/rsyslog.conf.j2 [hunk 1]
```diff
@@ -54,12 +54,10 @@ $outchannel {{ file_name }},/var/log/user/{{ file_name }},{{ file_options.archiv
 {%         endif %}
 {%         if host_options.protocol is vyos_defined('tcp') %}
 {%             if host_options.format.octet_counted is vyos_defined %}
-{{ tmp | join(';') }} @@(o){{ host_name | bracketize_ipv6 }}:{{ host_options.port }};RSYSLOG_SyslogProtocol23Format
-{%             else %}
-{{ tmp | join(';') }} @@{{ host_name | bracketize_ipv6 }}:{{ host_options.port }}
+{{ tmp | join(';') }} @@{{ '(o)' if host_options.format.octet_counted is vyos_defined }}{{ host_name | bracketize_ipv6 }}:{{ host_options.port }}{{ ';RSYSLOG_SyslogProtocol23Format' if host_options.format.include_timezone is vyos_defined }}
```
Comment (sever-sever) [inline data/templates/rsyslog/rsyslog.conf.j2:57]: As I remember it could be not only for TCP

Comment (c-po) [reply data/templates/rsyslog/rsyslog.conf.j2:57]: What exactly do you mean? Octet counting, or the message format?

Comment (sever-sever) [reply data/templates/rsyslog/rsyslog.conf.j2:57]: I mean this check ` is vyos_defined('tcp') `

```
{%         if host_options.protocol is vyos_defined('tcp') %}
{%             if host_options.format.octet_counted is vyos_defined %}
{{ tmp | join(';') }} @@{{ '(o)' if host_options.format.octet_counted is vyos_defined }}{{ host_name | bracketize_ipv6 }}:{{ host_options.port }}{{ ';RSYSLOG_SyslogProtocol23Format' if host_options.format.include_timezone is vyos_defined }}
```

But it is probably for another PR

**Review by sever-sever — APPROVED** [review]

**Comment by c-po** [comment]
@MergifyIo backport sagitta circinus

**Pull request merged**